olive/cli/capture_onnx.py

```diff
@@ -207,7 +226,7 @@ def get_run_config(self, tempdir: str) -> Dict:
 
 
 TEMPLATE = {
```

Comment:
not directly related to this PR since the template was already present. But wouldn't precision "fp16" with cpu ep fail for the model builder?

Reply:
Yeah fp16 doesn't work with cpu. There will be an error message saying this. But I updated the cli to support model builder gpu as well.

Reply:
the `device` argument is now being used for two different things. for conversion, it is the device where the model is placed on while exporting. for model builder, it is the target device to use the model on.

Reply:
Could also confirm, if just providing the device is fine in the accelerators? I think it might loop over both cpuep and cudaep for gpu device.

Reply:
1. Yeah this argument can be different meanings for different pass, as long as we don't run these 2 passes together. 
2. Yes it will for loop all available eps, which confuses me. I thought if this is a no search and no evaluator run, only one available ep should be picked up to run the workflow, cause Olive doesn't need metrics to compare anything. Is this expected?

Reply:
Yeah, Mike added the logic to automatically infer the ep from device. That was when we wanted to do multi-ep workflows: user provided target device, and we produce all valid ep models.

If we just provide a single ep, the device can be auto inferred.